src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java

```diff
@@ -0,0 +1,111 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.participation.Participation;
+import seedu.address.model.person.Attendance;
+import seedu.address.model.person.Person;
+import seedu.address.model.tutorial.Tutorial;
+
+/**
+ * Marks the attendance of a person identified using it's displayed index from the addres book as absent.
+ */
+public class UnmarkAttendanceByStudentCommand extends Command {
+
+    public static final String COMMAND_WORD = "umas";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Unmarks the attendance of the student identified "
+            + "by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_ATTENDANCE + "ATTENDANCE]"
+            + "[" + PREFIX_TUTORIAL + "TUTORIAL]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_ATTENDANCE + " 20/10/2024 "
+            + PREFIX_TUTORIAL + " Math";
+
+    public static final String MESSAGE_UNMARK_ATTENDANCE_STUDENT_SUCCESS =
+            "Unmarked attendance of %1$s student for %2$s tutorial on %3$s";
+
+    public static final String MESSAGE_INVALID_TUTORIAL_FOR_STUDENT =
+            "The student does not take %1$s tutorial";
+
+    private final Index targetIndex;
+    private final Attendance attendance;
+    private final Tutorial tutorial;
+
+    /**
+     * @param targetIndex Index of the person in the filtered person list to mark
+     * @param attendance Attendance of the person specified by index
+     * @param tutorial Tutorial the student attended
+     */
+    public UnmarkAttendanceByStudentCommand(Index targetIndex, Attendance attendance, Tutorial tutorial) {
+        requireAllNonNull(targetIndex, attendance, tutorial);
+        this.targetIndex = targetIndex;
+        this.attendance = attendance;
+        this.tutorial = tutorial;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person studentToUnmarkAttendance = lastShownList.get(targetIndex.getZeroBased());
+        Participation currentParticipation = model.getParticipationList().stream()
+                .filter(participation -> participation.getStudent().equals(studentToUnmarkAttendance)
+                        && participation.getTutorial().equals(this.tutorial)).findFirst()
+                .orElseThrow(() -> new CommandException(String.format(MESSAGE_INVALID_TUTORIAL_FOR_STUDENT, tutorial)));
+
+        List<Attendance> updatedAttendance = new ArrayList<>(currentParticipation.getAttendanceList());
+        updatedAttendance.add(attendance);
```

Comment:
Here should be removing attendance?

Reply:
Yea I think we should remove the attendance here and throw an exception if the attendance for that date is not found

Reply:
Thanks for the catch! Fixed now.